xwiki-platform-core/xwiki-platform-search/xwiki-platform-search-solr/xwiki-platform-search-solr-ui/src/main/resources/Main/SolrSearch.xml

```diff
@@ -198,14 +198,33 @@
     }
   };
 
-  var addFacetValueCheckbox = function() {
-    var checkBox = $(document.createElement('input')).attr('type', 'checkbox');
+  var addFacetValueCheckbox = function(index) {
+    // Create an id unique to the facet value.
+    let facetContainer = $(this).parents('.search-facet').first();
+    let checkBoxId = facetContainer.attr('data-name') + '-' + index.toString() + '-checkbox';
+
+    // Initialize the checkbox.
+    let checkBox = $(document.createElement('input')).attr('type', 'checkbox');
+    checkBox.attr('id', checkBoxId);
+
+    // Initialize the checkbox label.
+    let checkBoxLabel = $(document.createElement('label'));
+    checkBoxLabel.attr('for', checkBoxId);
+    checkBoxLabel.addClass('sr-only');
+    // We duplicate the content of the anchor in the facet value instead of having it be a label
+    // because we want to keep the use case "Open in a new page" possible.
+    $(this).contents().clone().appendTo(checkBoxLabel);
```

Comment:
According to https://dequeuniversity.com/rules/axe/4.6/label-title-only, using `aria-labelledby` should be sufficient.

Reply:
> Also, did you consider using a title attribute instead of the label? I'm not sure this would be better, I'm just wondering if it is really good to duplicate the link content.

See https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#accessiblenameguidancebyrole
They recommend adding a `label` to `type=checkbox` inputs without text content.

____
> I assume it is not sufficient to use an aria-labelledby attribute with a reference to the link? [...] 
According to https://dequeuniversity.com/rules/axe/4.6/label-title-only, using aria-labelledby should be sufficient.

The label is an HTML functionality, it's always better to use raw HTML than ARIA:
> If you can use a native HTML element [[HTML51](https://www.w3.org/TR/using-aria/#bib-html51)] or attribute with the semantics and behavior you require already built in, instead of re-purposing an element and adding an ARIA role, state or property to make it accessible, then do so.

From https://www.w3.org/TR/using-aria/#rule1

So if it's not too much trouble, I'd rather use a `label` than `aria-labelledby`. Usually in XWiki, it's more annoying because we need to generate unique ids, and it's not straightforward to get them.

___
> I'm just wondering if it is really good to duplicate the link content.

I wondered too, but it seems to me like it's necessary if we want the "false label" to be an anchor (and keep its 'Open in another tab' functionality)

See the first comment I added on the PR opening post (not clearly worded though x) ) :

> In order to keep the "Open in a new tab" functionality of the anchor, I chose not to replace it with a label like what I originally thought would be the good solution. [It's not possible to place an anchor inside a label](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label) and [it's not recommended to place a label inside of an anchor](https://stackoverflow.com/questions/21635431/wrap-inputs-label-in-hyperlink-anchor-tag). So I decided to duplicate the content in the anchor to put it in the label and give the label a sr-only (does not take visual space but is still active) class. See the ticket comments for more details.

Reply:
Have you checked how this is read in a screen reader? Won't the screen reader read the label twice now?

Reply:
Addressed in https://github.com/xwiki/xwiki-platform/pull/2187/commits/7c4005a71dc38a98189bd6d511099567b7bbc96a